conn_linux_test.go

```diff
@@ -269,9 +269,51 @@ func TestLinuxValidate(t *testing.T) {
 	}
 }
 
-func testLinuxConn(t *testing.T) (*conn, *testSocket) {
+func TestLinuxConnConfig(t *testing.T) {
+
+	tests := []struct {
+		name   string
+		config *Config
+		groups uint32
+	}{
+		{
+			name:   "Default Config",
+			config: &Config{},
+			groups: 0x0,
+		},
+		{
+			name:   "Config with Groups RTMGRP_IPV4_IFADDR",
+			config: &Config{Groups: 0x10},
+			groups: 16,
+		},
+		{
+			name:   "Config with Groups RTMGRP_IPV4_IFADDR | RTMGRP_IPV4_ROUTE",
+			config: &Config{Groups: 0x10 | 0x40},
+			groups: 80,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			_, s := testLinuxConn(t, tt.config)
+
+			if want, got := tt.groups, s.bind.(*syscall.SockaddrNetlink).Groups; want != got {
+				t.Fatalf("unexpected error:\n- want: %v\n-  got: %v",
+					want, got)
+			}
+
+		})
+	}
+
+}
+
+func testLinuxConn(t *testing.T, config *Config) (*conn, *testSocket) {
 	s := &testSocket{}
-	c, err := bind(s)
+	if config == nil {
```

Comment:
The problem with this is that right now, our test is testing `testLinuxConn`'s version of a `Conn` instead of `Dial`'s.  I think the most expedient approach is to push the `nil` check into `bind()` and handle it per-OS, unfortunately.  This being said, I doubt this package will ever work on other platforms so that's probably okay.